### ISSUE TITLE
Add option for supporting camera via pipewire

### ISSUE THREAD

**Issue by victornoel** [IsmaelMartinez/teams-for-linux#1528]
**Is your feature request related to a problem? Please describe.**

When using camera that relies on libcamera (the most famous is ipu6 cams from intel, cf https://jgrulich.cz/2024/12/13/when-your-webcam-doesnt-work-solving-firefox-and-pipewire-issues/ for some related context), chrome must be configured to use pipewire for the camera in one of its flags:
![image](https://github.com/user-attachments/assets/9e371e13-b086-4ca3-bfdf-c334b499c594)

**Describe the solution you'd like**

Ideally I would like to have a switch somewhere in the config (because I guess not everybody wants this feature) that would enable the flag.

**Describe alternatives you've considered**

Using chromium or firefox directly and not this app :)

**Additional context**

I worry that it's not possible to do this from teams-for-linux and that support must be added directly at the electron level?

**Comment by victornoel** [comment]
I went and created the corresponding ticket at electron: https://github.com/electron/electron/issues/45058

**Comment by IsmaelMartinez** [comment]
If it is a chromium flag that is supported by electron, then you can pass it as an option. 

https://github.com/IsmaelMartinez/teams-for-linux/tree/main/app/config

Check the electronCLIFlags option. You can see examples of its use in https://github.com/IsmaelMartinez/teams-for-linux/issues/1201

**Comment by victornoel** [comment]
Hey @IsmaelMartinez, that was indeed what I was thinking and why I created a ticket at electron.

They told we to use something called `app.commandLine.appendSwitch`, does it ring a bell and is there any way to specify that for teams-for-linux? see https://github.com/electron/electron/issues/45058#issuecomment-2552276693

Thanks!

**Comment by victornoel** [comment]
ha, apparently that is exactly what `electronCLIFlags` is for. Now I just need to validate that it's working or if something is missing.

**Comment by IsmaelMartinez** [comment]
glad you found the place. Do reach out at the matrix channel https://matrix.to/#/#teams-for-linux_community:gitter.im and I am sure others can share how to put those values. Sorry it took me a while to get back.

**Comment by LecrisUT** [comment]
@victornoel can you ELI5 what you mean with `electronCLIFlags`?

**Comment by victornoel** [comment]
@LecrisUT it is one of the available option in the teams-for-linux json config file to be able to pass options to Electron, see https://github.com/IsmaelMartinez/teams-for-linux/blob/main/app/config/README.md for some example! 

I hope this helps :)

**Comment by victornoel** [comment]
FYI I managed to enable the pipewire camera support on wayland with the following setting:
```json
{
	"electronCLIFlags": [
		["enable-features", "WebRtcPipeWireCamera"]
	]
}
```

Chrome is broken so I get no image, but the camera is actually detected, which is more than what I ever managed to do.

I'm using v2.0.0 of teams-for-linux (so a recent electron version).

**Comment by Ourouk** [comment]
> FYI I managed to enable the pipewire camera support on wayland with the following setting:
> 
> {
> 	"electronCLIFlags": [
> 		["enable-features", "WebRtcPipeWireCamera"]
> 	]
> }
> 
> Chrome is broken so I get no image, but the camera is actually detected, which is more than what I ever managed to do.
> 
> I'm using v2.0.0 of teams-for-linux (so a recent electron version).

Hey went to a similar rabbit hole, it's cool that u got it working but it's still strange with the config files but actually  
In index.js line 119->132
```js
function addCommandLineSwitchesAfterConfigLoad() {
  // Wayland
  if (process.env.XDG_SESSION_TYPE === "wayland") {
    console.info("Running under Wayland, switching to PipeWire...");

    const features = app.commandLine.hasSwitch("enable-features")
      ? app.commandLine.getSwitchValue("enable-features").split(",")
      : [];
    if (!features.includes("WebRTCPipeWireCapturer"))
      features.push("WebRTCPipeWireCapturer");

    app.commandLine.appendSwitch("enable-features", features.join(","));
    app.commandLine.appendSwitch("use-fake-ui-for-media-stream");
  }
```
We see that the features is already tried to be activated but doesn't seems to work the default way, the commit adding it was like 3 months ago and I don't understand why the injection trough setting would change something.
(Trying to enable the setting on vesktop and jitsi-meet-electron too)
